simd/src/Kokkos_SIMD_Scalar.hpp

```diff
@@ -42,13 +42,21 @@ class basic_simd_mask<T, simd_abi::scalar> {
   bool m_value;
 
  public:
-  using value_type                            = bool;
-  using simd_type                             = basic_simd<T, simd_abi::scalar>;
-  using abi_type                              = simd_abi::scalar;
-  KOKKOS_DEFAULTED_FUNCTION basic_simd_mask() = default;
+  using value_type = bool;
+  using simd_type  = basic_simd<T, simd_abi::scalar>;
+  using abi_type   = simd_abi::scalar;
+
   KOKKOS_FORCEINLINE_FUNCTION static constexpr std::size_t size() { return 1; }
```

Comment:
Two reminders: should be templated on bytes not T (the whole class, and size() should return integral_constant (not for this PR).

Reply:
Noted. I took many notes like this to find what's missing or inconsistent while making through this change.
I plan on making these changes in separate PRs subsequently.